src/libstd/sys/windows/mutex.rs

```diff
@@ -57,3 +58,35 @@ impl Mutex {
         // ...
     }
 }
+
+pub struct ReentrantMutex { inner: UnsafeCell<ffi::CRITICAL_SECTION> }
+
+unsafe impl Send for ReentrantMutex {}
+unsafe impl Sync for ReentrantMutex {}
+
+impl ReentrantMutex {
+    pub unsafe fn new() -> ReentrantMutex {
+        let section: ffi::CRITICAL_SECTION = mem::uninitialized();
+        ffi::InitializeCriticalSection(section);
+        ReentrantMutex {
+            inner: UnsafeCell { value: section }
```

Comment:
From [the documentation](https://msdn.microsoft.com/en-us/library/windows/desktop/ms683472%28v=vs.85%29.aspx):

> A critical section object cannot be moved or copied.

Perhaps this could have a separate `initialize` method as well?